xarray_schema/dataset.py

```diff
@@ -27,7 +28,7 @@ def __init__(
 
         self.data_vars = data_vars
         self.coords = coords
-        self.attrs = attrs
+        self.attrs = AttrSchema(attrs)
```

Comment:
what's the reason for needing to do this here, as opposed to in the setter?

Reply:
I've fixed this to use a setter

Reply:
By the way, have you considered using dataclasses https://docs.python.org/3/library/dataclasses.html or attrs classes https://www.attrs.org/en/stable/ to represent schemas?  They seem a natural fit.

Reply:
100% agree that we could transition to datacalsses. I may have even tried this in an earlier version. If memory serves, I ran into a problem with inheritance and decided to roll my own for the purposes of the MVP.

Reply:
If you'd like, we can discuss this idea in separate issue.